crates/router/src/types/api/customers.rs

```diff
@@ -1,16 +1,41 @@
 use api_models::customers;
 pub use api_models::customers::{CustomerDeleteResponse, CustomerId, CustomerRequest};
+use error_stack::ResultExt;
 use serde::Serialize;
 
-use crate::{newtype, types::storage};
+use crate::{
+    core::errors::{self, RouterResult},
+    newtype,
+    pii::PeekInterface,
+    types::domain,
+    utils::{self, ValidateCall},
+};
 
 newtype!(
     pub CustomerResponse = customers::CustomerResponse,
     derives = (Debug, Clone, Serialize)
 );
 
-impl From<storage::Customer> for CustomerResponse {
-    fn from(cust: storage::Customer) -> Self {
+pub(crate) trait CustomerRequestExt: Sized {
+    fn validate(self) -> RouterResult<Self>;
+}
+
+impl CustomerRequestExt for CustomerRequest {
+    fn validate(self) -> RouterResult<Self> {
+        self.email
+            .as_ref()
+            .validate_opt(|email| utils::validate_email(email.0.peek()))
+            .change_context(errors::ApiErrorResponse::InvalidDataFormat {
+                field_name: "email".to_string(),
+                expected_format: "valid email address".to_string(),
+            })?;
+
+        Ok(self)
+    }
+}
```

Comment:
This was removed in #819, I'm assuming this was added back during merge conflict resolution.